app/gui2/src/components/CreateNewNode.vue

```diff
@@ -0,0 +1,104 @@
+<script setup lang="ts">
+import SvgButton from '@/components/SvgButton.vue'
+import { useVisualizationConfig } from '@/providers/visualizationConfig'
+import { Ast } from '@/util/ast'
+import { Pattern } from '@/util/ast/match'
+import type { NodeCreationOptions } from './GraphEditor/nodeCreation'
+
+type SortDirection = 'asc' | 'desc'
+export type SortModel = {
+  columnName: string
+  sortDirection: SortDirection
+  sortIndex: number
+}
+
+const props = defineProps<{
+  filterModel: {
+    [key: string]: {
+      values: any[]
+      filterType: string
+    }
+  }
+  sortModel: SortModel[]
+}>()
+
+const config = useVisualizationConfig()
+
+const makeSortPattern = () => {
+  const sortMapping = {
+    asc: '..Ascending',
+    desc: '..Descending',
+  }
+  return props.sortModel
+    .filter((sort) => sort?.columnName)
+    .sort((a, b) => a.sortIndex - b.sortIndex)
+    .map((sort) => {
+      return `(..Name '${sort.columnName}' ${sortMapping[sort.sortDirection]})`
```

Comment:
This looks hazardous. Quotation marks and many other characters have special meaning inside an Enso interpolated text literal. The `Ast.TextLiteral.new` function performs any necessary escaping to create a text literal representing the given text.